encoders/src/main/scala/au/csiro/pathling/encoders/SerializerBuilder.scala

```diff
@@ -107,19 +108,42 @@ private[encoders] class SerializerBuilderProcessor(expression: Expression,
   }
 
   override def proceedCompositeChildren(value: CompositeCtx[Expression, (String, Expression)]): Seq[(String, Expression)] = {
-    dataTypeMappings.overrideCompositeExpression(expression, value.compositeDefinition)
-      .getOrElse(super.proceedCompositeChildren(value))
+
+    value.compositeDefinition.getImplementingClass match {
+      case cls if classOf[Quantity].isAssignableFrom(cls) =>
+        val valueExp = Invoke(expression, "getValue", ObjectType(classOf[java.math.BigDecimal]))
+        val codeExp = Invoke(expression, "getCode", ObjectType(classOf[java.lang.String]))
+        // TODO: Maybe create specialized UCUM functions returning spark.Decimal and UTF8String
+        // TODO: Maybe move to overrideCompositeExpression() providing it with a callback to generate default fields on request (a lazy call to super)
```

Comment:
Short term maybe just refactor our the creation of the extra serializers, 
e.g. .
```getOrElse(super.proceedCompositeChildren(value) ++  createSynhteticSerialzers(value))```

Longer term consider together with schema (see the comment in schema converter)
This can be done in `overrideCompositeExpression` it's signature is change to:
```
def overrideCompositeExpression(inputObject: Expression,
    definition: BaseRuntimeElementCompositeDefinition[_], 
    defaultFields:  => Seq[ExpressionWithName] // this is a lazy argument only evaluated it actually used
     ):Option[Seq[ExpressionWithName]]
```

Reply:
Refactored.

Long term solution captured in https://github.com/aehrc/pathling/issues/842 [Review and harmonize encoding customisations]